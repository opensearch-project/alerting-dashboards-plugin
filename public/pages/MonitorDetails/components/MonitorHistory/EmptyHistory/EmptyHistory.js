/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import PropTypes from 'prop-types';
import { EuiSmallButton, EuiEmptyPrompt, EuiText } from '@elastic/eui';

const EmptyHistory = ({ onShowTrigger }) => (
  <EuiEmptyPrompt
    style={{ maxWidth: '45em' }}
    body={
      <EuiText size="s">
        <p>
          There are no triggers. Create a trigger to start alerting. Once an alarm is triggered, the
          state will be displayed in time series.
        </p>
      </EuiText>
    }
    actions={
      <EuiSmallButton fill onClick={onShowTrigger}>
        Edit monitor
      </EuiSmallButton>
    }
  />
);

EmptyHistory.propTypes = {
  onShowTrigger: PropTypes.func.isRequired,
};

export default EmptyHistory;
