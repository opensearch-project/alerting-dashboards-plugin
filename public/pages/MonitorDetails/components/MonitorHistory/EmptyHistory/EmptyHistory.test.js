/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import { render } from 'enzyme';

import EmptyHistory from './EmptyHistory';

describe('<EmptyHistory/>', () => {
  test('renders', () => {
    expect(render(<EmptyHistory onShowTrigger={jest.fn()} />)).toMatchSnapshot();
  });
});
