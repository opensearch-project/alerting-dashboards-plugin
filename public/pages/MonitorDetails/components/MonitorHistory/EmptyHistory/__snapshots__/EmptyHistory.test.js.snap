// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EmptyHistory/> renders 1`] = `
<div
  class="euiEmptyPrompt"
  style="max-width:45em"
>
  <span
    class="euiTextColor euiTextColor--subdued"
  >
    <div
      class="euiText euiText--medium"
    >
      <div
        class="euiText euiText--small"
      >
        <p>
          There are no triggers. Create a trigger to start alerting. Once an alarm is triggered, the state will be displayed in time series.
        </p>
      </div>
    </div>
  </span>
  <div
    class="euiSpacer euiSpacer--l"
  />
  <button
    class="euiButton euiButton--primary euiButton--small euiButton--fill"
    type="button"
  >
    <span
      class="euiButtonContent euiButton__content"
    >
      <span
        class="euiButton__text"
      >
        Edit monitor
      </span>
    </span>
  </button>
</div>
`;
